pkg/create/prompts.go

```diff
@@ -0,0 +1,90 @@
+package create
+
+import (
+	"fmt"
+	"regexp"
+	"slices"
+
+	"github.com/AlecAivazis/survey/v2"
+)
+
+func validateAppName(val any) error {
+	str, ok := val.(string)
+	if !ok {
+		return fmt.Errorf("invalid input type")
+	}
+
+	if len(str) == 0 {
+		return fmt.Errorf("project name cannot be empty")
+	}
+
+	// Validate project name: only letters, numbers, underscores, and hyphens
+	matched, err := regexp.MatchString(`^[A-Za-z\-_\d]+$`, str)
+	if err != nil {
+		return err
+	}
+	if !matched {
+		return fmt.Errorf("project name may only include letters, numbers, underscores, and hyphens")
+	}
+	return nil
+}
+
+func PromptForAppName(providedAppName string) (string, error) {
+	if providedAppName != "" {
+		return providedAppName, nil
```

Comment:
Consider also validating when the app name is provided via flag. Currently `--name "invalid name!"` would bypass the validation logic.